worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/FileWriter.java

```diff
@@ -294,10 +286,12 @@ public long close() throws IOException {
     return bytesFlushed;
   }
 
-  public void destroy() {
+  public synchronized void destroy(IOException ioException) {
```

Comment:
destroy do not always means exception. Common cleanup should not be considered exception. Anyway it's fine since the destroyed file should never be read.

Reply:
Do we need to add an overloaded method which don't need exception argument?

Reply:
Currently we don't distinguish normal from unnormal cleanup. I think we can do it in later prs.